source/user/df.rst

```diff
@@ -5,3 +5,143 @@ Density fitting
+    mydf = df.DF(mol).build()
+    mf = scf.RHF(mol).density_fit()
+    mf.with_df = mydf
+    mf.kernel()
+
+Setting :attr:`with_df` to ``None`` turns DF off.
+
+Once the SCF calculations using DF are done, all post-SCF methods will use DF automatically. The example below performs a TDDFT/TDA calculation using DF::
+
+    from pyscf import gto, dft, tddft
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvp')
+    mf = dft.RKS(mol, xc="b3lyp").density_fit().run()
+    td = tddft.TDA(mf).run()
+    print(td.e)
+
+
+.. _choice_of_auxbasis:
+
+Choice of auxiliary basis
+=========================
+
+General consideration
+---------------------
+
+The choice of auxiliary basis depends on the AO basis.
+By default, PySCF uses the pre-defined auxiliary basis set optimized for the AO basis set if the former exists.
+This includes many commonly used AO basis sets in electronic structure calculations, e.g., the Ahlrichs' `def2` family, :cite:`Hellweg07TCA,Weigend98CPL` the Dunning's `cc` family, :cite:`Weigend02PCCP` etc.
+
+When pre-defined auxiliary basis set is not available, an even-tempered basis (ETB) set is generated by the following rule
+
+.. math::
+
+    \varphi &= r^l \exp(-\zeta_{il} r^2), \quad i = 0, 1, \cdots, n \\
+    \zeta_{il} &= \alpha_l \times \beta^i
+
+where :math:`\alpha_l` and :math:`N` are inferred automatically from the AO basis, and :math:`\beta = 2.0` by default.
+
+The user can overwrite the default choice by setting :attr:`auxbasis` upon initialization or at a later stage::
+
+    from pyscf import gto, scf, df
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='cc-pvdz')
+    mf = scf.RHF(mol).density_fit(auxbasis="weigend")
+    mf.kernel() # -108.910953335055
+    mf.with_df.auxbasis = "cc-pvdz-jkfit"   # this is the default
+    mf.kernel() # -108.913710743723
+    mf.with_df.auxbasis = df.aug_etb(mol, beta=1.7) # ETB with beta = 1.7
+    mf.kernel() # -108.914059329528
+
+The user can check what auxiliary basis set is used at any time by::
+
+    print(mf.with_df.auxmol.basis)
+
+More examples on using customized auxiliary basis can be found in :source:`examples/df/01-auxbasis.py`.
+
+
+Special consideration for DFT
+-----------------------------
+
+For DFT calculations with pure exchange-correlation functionals (i.e., LDA and GGA), the default auxiliary basis, which is designed for fitting both the Coulomb and the Hartree-Fock exchange integrals, may be unnecessarily too large. We recommend using the ``weigend`` family :cite:`Weigend06PCCP` for a more cost-effective choice as shown in the following example::
+
+    from pyscf import gto, dft
+    mol = gto.M(atom='N 0 0 0; N 0 0 1.2', basis='def2-tzvpp')
+    mf = dft.RKS(mol, xc="pbe").density_fit().run() # -109.432329411505
+    print(mf.with_df.auxmol.basis)                  # default: def2-tzvpp-jkfit
+    print(mf.with_df.auxmol.nao_nr())               # 154 aux basis functions
+    mf.with_df.auxbasis = "weigend"
```

Comment:
"weigend" should be `def2-universal-jfit`

Reply:
and one can note that the alternative for calculations including exact exchange is `def2-universal-jkfit`

Reply:
Changed to `def2-universal-jfit` and added a note that this is same as using `weigend`.
For the `def2-universal-jkfit`, I think that is the default used by PySCF if AO basis from the `def2` family is used.